src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -31,6 +31,12 @@ public class RestoreCommand
         private readonly Dictionary<RestoreTargetGraph, Dictionary<string, LibraryIncludeFlags>> _includeFlagGraphs
             = new Dictionary<RestoreTargetGraph, Dictionary<string, LibraryIncludeFlags>>();
 
+        public KeyValuePair<string, Guid> ParentId { get; }
+
+        public KeyValuePair<string, Guid> RestoreId = new KeyValuePair<string, Guid>(nameof(RestoreId), Guid.NewGuid());
+
+        public const string ProjectRestoreInformation = "ProjectRestoreInformation";
```

Comment:
private

Reply:
also in case, the parent want to store child's event name in their table